odl/discr/discr_ops.py

```diff
@@ -130,110 +140,215 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
 
     method, method_in = str(method).lower(), method
     if method not in ('central', 'forward', 'backward'):
-        raise ValueError("method '{}' has to be central, forward or backward"
+        raise ValueError('method {} is not central, forward or backward'
                          ''.format(method_in))
 
+    if isinstance(padding, (int, float)) or padding in (None, 'replicate'):
+        if isinstance(padding, int):
+            padding = float(padding)
+    else:
+        raise ValueError('padding value ({}) not valid'.format(padding))
+
+    if edge_order is None:
+        if method == 'central':
+            edge_order = 2
+        else:
+            edge_order = 1
+    else:
+        if edge_order not in [1, 2]:
+            raise ValueError('edge order ({0}) not valid'.format(edge_order))
+
+    if f_arr.shape[axis] == 2 and edge_order == 2:
+        raise ValueError('shape ({0}) of array to small to use one-sided '
+                         'second-order accurate edges'.format(f_arr.shape))
+
+    if padding is not None:
+        if method == 'central' and edge_order == 1:
+            raise ValueError(
+                'central differences with padding only use second-order '
+                'accuracy at edges. Edge-order accuracy can only be '
+                'triggered without padding for one-sided differences at '
+                'edges')
+        if method in ('forward', 'backward') and edge_order == 2:
+            raise ValueError(
+                'forward/backward difference with padding use the same '
+                'accuracy on edges as in the interior. Edge-order accuracy '
+                'can only be triggered without padding for one-sided '
+                'differences at edges.')
```

Comment:
A bit long, these messages. Just say "cannot combine x with y", users can check the documentation (it should explain the valid combinations, of course!)